target/linux/kirkwood/image/Makefile

```diff
@@ -143,6 +143,14 @@ define Device/zyxel_nsa310b
 endef
 TARGET_DEVICES += zyxel_nsa310b
 
+define Device/zyxel_nsa310s
+  DEVICE_VENDOR := ZyXEL
+  DEVICE_MODEL := NSA310s
+  DEVICE_PACKAGES := kmod-gpio-button-hotplug
```

Comment:
Kernel module for Ethernet port is probably missing here.

Reply:
It is built-in.

Reply:
Indeed, ack.